core/src/main/java/org/elasticsearch/rest/action/admin/indices/analyze/RestAnalyzeAction.java

```diff
@@ -129,7 +130,18 @@ public static void buildFromContent(BytesReference content, AnalyzeRequest analy
                             }
                             charFilters.add(parser.text());
                         }
-                        analyzeRequest.tokenFilters(charFilters.toArray(Strings.EMPTY_ARRAY));
+                        analyzeRequest.charFilters(charFilters.toArray(new String[charFilters.size()]));
+                    } else if ("detail".equals(currentFieldName) && token == XContentParser.Token.VALUE_BOOLEAN) {
```

Comment:
Can we use the ParseFieldMatcher and ParseField objects here since we should be moving all parsers over to that anyway?